### ISSUE TITLE
Add support for pop-up menus

### ISSUE THREAD

**Issue by wrc-marek** [masaccio/numbers-parser#74]
Is there a way to add a pop-up menu?

<img width="368" alt="Screenshot 2024-02-22 at 22 41 31" src="https://github.com/masaccio/numbers-parser/assets/98819454/beb4bf67-0060-4ccd-9c2b-3caf98f3527b">

I have tried to open a Numbers file with a pop-up menu, and after saving the pop-up is removed:

<img width="341" alt="Screenshot 2024-02-22 at 22 42 24" src="https://github.com/masaccio/numbers-parser/assets/98819454/ef3d97a0-31b5-42ae-9382-f2fe8b5ca16b">

**Comment by masaccio** [comment]
This has been published in v4.9.0. There's an example [in this test](https://github.com/masaccio/numbers-parser/blob/main/tests/test_formatting.py#L881). Error handling is a bit ropey at the moment. For example there's no check that values written match the available popup values. You can mix types in the menu, but currently only `float` and `str` are supported. Numbers detects percentages and currencies, but I think that will be fragile in python.

**Comment by wrc-marek** [comment]
It's working thanks. The only problem that I see is that it's also removing pop-up menu when the value is `None`.
Original file:
<img width="109" alt="Screenshot 2024-02-25 at 19 38 46" src="https://github.com/masaccio/numbers-parser/assets/98819454/873eadd6-0cba-46d6-ab70-2b244268eb2e">
After saving:
<img width="391" alt="Screenshot 2024-02-25 at 19 39 35" src="https://github.com/masaccio/numbers-parser/assets/98819454/5a3face8-89e9-4e61-a6c9-2289543283b6">

**Comment by masaccio** [comment]
As in you have a popup with an empty first value, load the document in `numbers-parser`, re-save and the resulting popup is removed?

**Comment by wrc-marek** [comment]
Correct, I have a document with many popups. Some popups have a non-empty default value and some have default value `None`. After loading this document into `number-parser` and re-saving, it keeps popups where the first value isn't empty but for those where the default value is `None` it removes popup definition and the cell ends up with empty value.

**Comment by masaccio** [comment]
v4.9.2 adds support for empty cells with popup menus.

**Comment by wrc-marek** [comment]
Works perfectly!

**Comment by wrc-marek** [comment]
I have one more problem with pop-ups when reading existing Numbers file. Here is how to reproduce the error:

1) Create a new Numbers file and save it to `template.numbers` file
2) Execute
```python
from numbers_parser import Document

doc = Document("./template.numbers")
table = doc.sheets[0].tables[0]
table.write(0, 1, "Dog")
table.set_cell_formatting(0, 1, "popup", popup_values=["Cat", "Dog", "Rabbit"], allow_none=True)
```

Error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "test.py", line 22, in <module>
    table.set_cell_formatting(0, 1, "popup", popup_values=["Cat", "Dog", "Rabbit"], allow_none=True)
  File "venv/lib/python3.11/site-packages/numbers_parser/document.py", line 1384, in set_cell_formatting
    self._set_cell_data_format(row, col, format_type, **kwargs)
  File "venv/lib/python3.11/site-packages/numbers_parser/document.py", line 1426, in _set_cell_data_format
    control_id = self._model.control_cell_archive(self._table_id, format_type, format)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.11/site-packages/numbers_parser/model.py", line 418, in control_cell_archive
    popup_id = self.cell_popup_model(self._control_specs.id(table_id), format)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "venv/lib/python3.11/site-packages/numbers_parser/model.py", line 161, in id
    return self._datalists[table_id]["id"]
           ~~~~~~~~~~~~~~~^^^^^^^^^^
KeyError: 904538
```

Tested on `numbers-parser==4.10.0`

**Comment by wrc-marek** [comment]
I tested also without `template.numbers` file and I'm also getting error. This doesn't work:

```python
doc = Document()
table = doc.sheets[0].tables[0]

table.write(1, 0, "Dog")
table.set_cell_formatting(1, 0, "popup", popup_values=["Cat", "Dog", "Rabbit"], allow_none=True)
```

This works:
```python
doc = Document()
table = doc.sheets[0].tables[0]

table.write(0, 0, False)
table.set_cell_formatting(0, 0, "tickbox")

table.write(1, 0, "Dog")
table.set_cell_formatting(1, 0, "popup", popup_values=["Cat", "Dog", "Rabbit"], allow_none=True)
```

**Comment by masaccio** [comment]
Thanks for spotting that using a different control before the popup hid the bug -- the popup code was not calling an init method.

I've added checks for empty cells and popups. Empty values are only allowed if `allow_none` is `True`:

``` python
    table.write(0, 0, "")
    # Valid
    table.set_cell_formatting(0, 0, "popup", popup_values=["Cat", "Dog", "Rabbit"], allow_none=True)
    # raises IndexError
    table.set_cell_formatting(0, 0, "popup", popup_values=["Cat", "Dog", "Rabbit"])
```

All published in v4.10.1.

**Comment by wrc-marek** [comment]
Works perfect, thanks!
